### DISCUSSION TITLE
How to make markers enabled when startup without toggling?

### DISCUSSION THREAD

**Discussion by failable** [kovidgoyal/kitty#4210] [General]
I know I can start the terminal, and press some keys like `f1` to enable markers. Can I make makers enabled by default? I'm pretty sure my markers do not harm performance too much. Thanks.

**Comment by page-down** [comment]
This can be achieved using watcher, and here is an example.

```python
from typing import Any, Dict

from kitty.typing import BossType
from kitty.window import Window


def on_focus_change(boss: BossType, window: 'Window', data: Dict[str, Any]) -> None:
    if data.get('focused'):
        window.set_marker('text 1 ERROR')                  
    else:
        window.remove_marker()
```

https://sw.kovidgoyal.net/kitty/invocation/#cmdoption-kitty-watcher
https://sw.kovidgoyal.net/kitty/launch/#cmdoption-launch-watcher

In the current unreleased version (try with the nightly version), the global watcher can be defined in the configuration file.

Or use launch to open a new window with marker enabled by default.

```text
launch --marker "text 1 ERROR"
```

https://sw.kovidgoyal.net/kitty/launch/#cmdoption-launch-marker

Or configure the default window to enable marker in the startup session file.

startup_session.conf
```text
launch --marker "text 1 ERROR"
```
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.startup_session

Or use remote control

.bashrc
```
kitty @ create-marker text 1 ERROR
```

https://sw.kovidgoyal.net/kitty/remote-control/#kitty-create-marker

**Reply by failable** [reply]
Hi, thanks for the solutions. I tried the session file way and it works for `text`, but does not seem to work with `regex` or `iregex` like `launch --marker "regex 1 \\bERROR\\b"`. Am I missing something here?

**Reply by failable** [reply]
The `.bashrc` way works perfectly. But I'd like to keep kitty configs separatly.

**Reply by page-down** [reply]
You need to escape correctly, for example the following is fine.

```text
launch --marker 'regex 1 \\bERROR\\b'
```

**Reply by failable** [reply]
Thanks, works like a charm.

**Reply by failable** [reply]
Hi, I have one more question. 

Is there a way to have the same markers work for startup session, new tabs, and new windows without writing the same `--marker` many times and binding different keys like `map f12 launch --type=tab --marker ...` and `map f11 launch --marker ...` in `kitty.conf` or `launch --marker ...` in session file?

**Reply by page-down** [reply]
As far as I know, no. You can define the `map ...` using a separate configuration file, with `include /path/to/marker-shortcuts.conf` in kitty.conf. Then use a script to generate the configuration file and the session file.

**Comment by failable** [comment]
Thanks for the advice, I will try it. I've thought that I could put some configs to startup session to achieve this, but didn't seem to find the right API for it.

**Reply by page-down** [reply]
@liebkne Please check my latest example. However only for windows that are opened by default.

**Comment by kovidgoyal** [comment]
IIRC you can use remote control for that, just add the remote control
command to your shell rc files.
